app/controllers/v0/account_controls_controller.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+module V0
+  class AccountControlsController < ApplicationController
+    skip_before_action :authenticate, :verify_authenticity_token
+    before_action :authenticate_service_account
+
+    VALID_CSP_TYPES = %w[logingov idme dslogon mhv].freeze
+
+    def csp_index
+      raise Common::Exceptions::ParameterMissing, 'icn' if params[:icn].blank?
+
+      serialized_user_verifications = serialize_user_verifications(user_verifications: fetch_verifications_by_icn)
+      Rails.logger.info('UserAccount CSP Index', { icn: params[:icn],
+                                                   requested_by: @service_account_access_token.user_identifier })
+      render json: { data: serialized_user_verifications }
+    rescue ActiveRecord::RecordNotFound
+      render json: { error: "User CSP Index not found. ICN:#{params[:icn]}" }, status: :not_found
+    end
+
+    def csp_lock
+      validate_csp_params
+
+      if user_verification.update(locked: true)
```

Comment:
I would do `user_verification.update!` and then rescue `RecordInvalid` for the 'lock failed' case

Reply:
It doesn't really matter but I think the more 'railsy' way is to use the conditional. `.update` will return true or false even if there are errors. If there are errors, it automatically rescues them and puts it in the `errors` attribute on the object. 

e.g. if this failed `user_verification.update(locked: true)` you can get the errors from `user_verification.errors.messages`. 

Again, doesn't really matter just FYI